ESP_Code4.3/ESP_Code/ESP_Code.ino

```diff
@@ -0,0 +1,817 @@
+             s.replace("@@SHARES@@", String(share_count));
+             s.replace("@@NODE@@", String(node_id));
+             
+             #if defined(ESP8266)
+                 s.replace("@@DEVICE@@", "ESP8266");
+             #elif defined(CONFIG_FREERTOS_UNICORE)
+                 s.replace("@@DEVICE@@", "ESP32-S2/C3");
+             #else
+                 s.replace("@@DEVICE@@", "ESP32");
+             #endif
+             
+             s.replace("@@ID@@", String(RIG_IDENTIFIER));
+             s.replace("@@MEMORY@@", String(ESP.getFreeHeap()));
+             s.replace("@@VERSION@@", String(SOFTWARE_VERSION));
+
+             #if defined(CAPTIVE_PORTAL)
+                 s.replace("@@RESET_SETTINGS@@", "&bull; <a href='/reset'>Reset settings</a>");
+             #else
+                 s.replace("@@RESET_SETTINGS@@", "");
+             #endif
+
+             #if defined(USE_DS18B20)
+                 sensors.requestTemperatures(); 
+                 float temp = sensors.getTempCByIndex(0);
+                 s.replace("@@SENSOR@@", "DS18B20: " + String(temp) + "*C");
+             #elif defined(USE_DHT)
+                 float temp = dht.readTemperature();
+                 float hum = dht.readHumidity();
+                 s.replace("@@SENSOR@@", "DHT11/22: " + String(temp) + "*C, " + String(hum) + "rh%");
+             #elif defined(USE_HSU07M)
+                 float temp = read_hsu07m();
+                 s.replace("@@SENSOR@@", "HSU07M: " + String(temp) + "*C");
+             #elif defined(USE_INTERNAL_SENSOR)
+                 float temp = 0;
+                 temp_sensor_read_celsius(&temp);
+                 s.replace("@@SENSOR@@", "CPU: " + String(temp) + "*C");
+             #else
+                 s.replace("@@SENSOR@@", "None");
+             #endif
+                 
+             server.send(200, "text/html", s);
+        }
+    #endif
+
+} // End of namespace
+
+MiningJob *job[CORE];
+
+#if CORE == 2
+  EasyFreeRTOS32 task1, task2;
+#endif
+
+void task1_func(void *) {
+    #if defined(ESP32) && CORE == 2
+      VOID SETUP() { }
+
+      VOID LOOP() {
+        job[0]->mine();
+
+        #if defined(DISPLAY_SSD1306) || defined(DISPLAY_16X2) || defined(DISPLAY_114)
+           float hashrate_float = (hashrate+hashrate_core_two) / 1000.0;
+           float accept_rate = (accepted_share_count / 0.01 / share_count);
+           
+           long millisecs = millis();
+           int uptime_secs = int((millisecs / 1000) % 60);
+           int uptime_mins = int((millisecs / (1000 * 60)) % 60);
+           int uptime_hours = int((millisecs / (1000 * 60 * 60)) % 24);
+           String uptime = String(uptime_hours) + "h" + String(uptime_mins) + "m" + String(uptime_secs) + "s";
+           
+           float sharerate = share_count / (millisecs / 1000.0);
+          
+           display_mining_results(String(hashrate_float, 1), String(accepted_share_count), String(share_count), String(uptime), 

```

Comment:
From our discussion on Discord:
I think I know, since it refers to a mutex it must be happening in a point when two threads found the share at almost or exactly the same time and both want to take control of the display? but it's weird as only one thread should be updating it unless I made something wrong or don't remember it correctly, I'll check in a second
No, indeed this is possible, both cores update the display. Can you check if that's the case by removing one of two display_mining_results calls?

Reply:
I had the same idea just 14 minutes ago, and removing the || defined(DISPLAY_114) from the second core seems to do the trick...
After seeing a few times the display half frozen before the console crashes...
it is now working since 15 minutes, but of course updates only on core 0 shares...
Maybe the solution would be to raise a flag while the display is still doing something, and check it in the mining loop (core 0 and core 1) to wait before sending the result ?